Using https instead of http to fetch Ipopt

adrienbernede opened the pull request:
It seems to me it wouldn’t hurt to use https there. As a matter of fact, http does not work for me, while https does.

@goxberry suggested we could go a bit further and now use archives available on GitHub:
https://github.com/coin-or/Ipopt/archive/releases/x.y.z.tar.gz.
This is also ready to be added to this PR if maintainers (@svigerske? @tkralphs?) agree.

adrienbernede committed:
Using https instead of http to fetch Ipopt

adamjstewart commented:
While I have you here, would any of you like to be listed as official maintainers of this Spack package? You don't have to be Spack experts, just someone willing to review PRs or take a look at bug reports for this package. If so, we can add:
```python
maintainers = ['foo', 'bar', ...]
```
with any of your GitHub handles.

adrienbernede committed:
Ipopt: moving to archives hosted on github

adrienbernede commented:
As a suggestion, this commit changes URLs to point to archives hosted on GitHub.
It also drops versions older that 3.12.6 at the moment, but extra "coverage" could be added.

adrienbernede commented:
@adamjstewart I am not ignoring your here, I just don’t think I should be maintainer of a project I don’t know... at all.

adrienbernede committed:
Fix: style

tkralphs commented:
Interesting, I wasn't aware of spack, thanks for pointing me to it. It could almost replace [coinbrew](https://github.com/coin-or/coinbrew), but not quite. I think @svigerske would be a better choice than me for listing as a maintainer for Ipopt, but if there are other COIN-OR packages, I could probably help out a bit. There are quite a few package managers out there with recipes for COIN-OR projects, though, so I'm trying to figure out where to allocate my limited bandwidth. 

Anyway, while I'm here, I noticed that the home page for Ipopt should be updated to https://github.com/coin-or/Ipopt.

adrienbernede committed:
Ipopt: Updating homepage

adamjstewart commented:
@tkralphs currently we have `adol-c`, `clp`, `cppad`, `csdp`, and `ipopt` packaged in Spack. If you're or anyone else are interested in being listed as a maintainer on any of those, let me know and I can add you. If there are any other COIN-OR projects you are interested in packaging in Spack, let me know and I can help.

svigerske commented:
I'm not sure what I have to agree on, but using the tarballs from GitHub should be ok.
Source tarballs are also still available at http://www.coin-or.org/download/source/Ipopt/ and I think they are plans to continue that, but the primary source could be the GitHub page.

adrienbernede committed:
Revert "Ipopt: moving to archives hosted on github"

This reverts commit 872037d286b6503c4d40eb3d8f40564e67b44042.

adrienbernede committed:
Ipopt: Back to previous tarball location.

Pagination on Github prevent spack from easily parse all available
versions. Also, due to recent migration to GitHub, tarballs for
versions up to 3.12.13 have be regenerated, changing the hash.

The current URL will apparently be supported, so we keep it, and give
the alternative one as a comment.

adamjstewart commented on var/spack/repos/builtin/packages/ipopt/package.py:
```diff
@@ -9,20 +9,28 @@
 
+    homepage = "https://github.com/coin-or/Ipopt"
+    url      = "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.13.2.tgz"
+    #Alternative: url      = "https://github.com/coin-or/Ipopt/archive/releases/3.13.2.tar.gz"
```
You can use:
```python
urls = [
    "https://github.com/coin-or/Ipopt/archive/releases/3.13.2.tar.gz",
    "https://www.coin-or.org/download/source/Ipopt/Ipopt-3.13.2.tgz",
]
```
and assuming they have the same checksums, Spack will try the first one, and if there are any download issues try the second one.

adrienbernede replied:
Precisely, they don’t all have the same checksum.

adrienbernede replied:
Let me paste here the last commit message:
"Pagination on Github prevent spack from easily parse all available
versions. Also, due to recent migration to GitHub, tarballs for
versions up to 3.12.13 have be regenerated, changing the hash.

The current URL will apparently be supported, so we keep it, and give
the alternative one as a comment."

adrienbernede replied:
That’s not really beautiful, I confess.

adrienbernede committed:
Fix: style

alalazo commented:
@adrienbernede What is the status of this PR?

adrienbernede committed:
Run checksum to update with latest version

adrienbernede commented:
> @adrienbernede What is the status of this PR?

Thanks for resurrecting this. On my side it looks like I did what I could do.

I just updated the PR with the second to latest version as detected by `spack checksum ipopt` (namely 3.13.3)
However, they also uploaded the 3.13.4 tarball, but with a change in the extension: `tgz` -> `tar.gz`. Spack does not seem to find it because of that.

I assume we could let them know while merging this as it is in a ready state.

alalazo commented:
@adrienbernede Yeah, let's proceed like that. Apologies for the time it took to merge this :sweat_smile:

adrienbernede commented:
No worries, it was never clearly stated that it was ready ;)
